### PR TITLE
Feature/chunk by memory

### PR DESCRIPTION
**Pull request by nbc**
Hi,

New PR to add a chunk_memory_size argument to table_to_parquet to split output file par memory size.

### CODE REVIEW

**Comment by ddotta** [comment]
Very good idea this `chunk_memory_size` argument I had thought about it at first then gave up...  

I had stuck with the idea of separating very large tables into chunks based on the number of rows (see [here](https://github.com/ddotta/parquetize/blob/main/vignettes/aa-conversions.Rmd#L28-36) in the vignette).  
But the disadvantage with my initial method is that you have to know "a priori" the number of lines in the very large file to convert...  

The only inconvenience I see with the `chunk_memory_size` argument is, for example :  
```
table_to_parquet(
  path_to_table = system.file("examples","iris.sas7bdat", package = "haven"),
  path_to_parquet = "Data_test",
  by_chunk = TRUE,
  chunk_memory_size = 5 /1024,
)
```

is that it returns :

✔ The SAS file is available in parquet format under Data_test/iris1-89.parquet
✔ The SAS file is available in parquet format under Data_test/iris90-178.parquet

which is wrong for iris90-178 which does not contain 89 lines but only 61.  

I think we need to change [this line](https://github.com/ddotta/parquetize/blob/main/R/utilities.R#L35)

File: R/table_to_parquet.R [hunk 1]
```diff
@@ -62,8 +66,28 @@
...
+#' )
+#'
+#' # Reading SPSS file by chunk (using `chunk_memory_size` argument)
+#' # and conversion to multiple parquet files :
+#'
+#' table_to_parquet(
+#'   path_to_table = system.file("examples","iris.sav", package = "haven"),
+#'   path_to_parquet = tempdir(),
+#'   by_chunk = TRUE,
+#'   chunk_memory_size = 5 /1024,
```
Comment (ddotta) [inline R/table_to_parquet.R:86]: @nbc 
Can you explain the use of 5/1024 ?
Do we add a recommendation on how to fill in the `chunk_memory_size` argument ?

File: tests/testthat/_snaps/table_to_parquet.md
Comment (ddotta) [inline tests/testthat/_snaps/table_to_parquet.md:1]: Ok, I didn't see you've already updated the vignette 👍

**Comment by nbc** [comment]
> heavy_check_mark The SAS file is available in parquet format under Data_test/iris1-89.parquet heavy_check_mark The SAS file is available in parquet format under Data_test/iris90-178.parquet
> 
> which is wrong for iris90-178 which does not contain 89 lines but only 61.
> 
> I think we need to change [this line](https://github.com/ddotta/parquetize/blob/main/R/utilities.R#L35)

Indeed.  Fixed in last push

File: R/table_to_parquet.R (hunk 1 above)
Comment (nbc) [reply R/table_to_parquet.R]: Done.

Comment (ddotta) [reply R/table_to_parquet.R]: Thanks @nbc !

File: vignettes/aa-conversions.Rmd [hunk 2]
```diff
@@ -29,13 +29,53 @@ When handling very large files, the risk that frequently occurs is that the R se
 This risk is even more present if you work locally on your computer and it can be limited if you work on remote servers.  
 `table_to_parquet()` offers this solution which answers a need expressed by parquetize users.  
 
-**The idea is to split the very large table into "chunks" based on the number of rows in the table in order to be able to simultaneously :**  
+**The idea is to split the very large table into "chunks" based on memory consumption of input data or on the number of rows in the table in order to be able to simultaneously :**  
 - **read a chunk of the very large database**  
 - **write this chunk in the floor file**
 
 Tip: The number of lines that each chunk must contain must be supported by the RAM of your computer/server. Ideally, the number of chunks to be defined must be limited. It should be in tens and not hundreds to limit the number of intermediate files (see example below).
```
[the remark this answers is not in the document]
Comment (ddotta) [reply vignettes/aa-conversions.Rmd]: You're right, tip moved 😃

**Pull request merged**